### PR TITLE
Rest Client - support injection for @RegisterClientHeaders

### LINKED ISSUE #9301: Microprofile rest client - Can't inject config/beans to clientheaderfactory also in 1.4.2.Final
**Describe the bug**
In issue #7531 a solution was given for injecting providers in microprofile rest client using the annotations @RegisterProvider or @RegisterClientHeaders. Unfortunately the fix doesn't seem to work for @RegisterClientHeaders and my injected beans are always null.

**Expected behavior**
A class that implements ClientHeadersFactory and used in @RegisterClientHeaders annotation should have it's injectable beans injected.

**Actual behavior**
Any field marked with @Inject is always null. @RegisterProvider works as intended.

**To Reproduce**
1. Clone project from marcinczeczko from [@here](https://github.com/marcinczeczko/quarkus-rest-providers-reproducer) (same as #7531)
2. Update quarkus to 1.4.2.Final in the pom.xml (both plugin and platform properties)
3. Start project and do GET request to localhost:8080/echo resource
4. Check log to see that Provider has proper injection and the HeaderFactory does not

**Configuration**
Using properties from repo.

**Screenshots**
No screenshot, but a part of the log using above reproduction steps;
```log
2020-05-14 15:54:41,126 INFO  [org.acm.res.MyHeadersFactory] (executor-thread-1) SomeBean = <null>
2020-05-14 15:54:41,126 INFO  [org.acm.res.MyHeadersFactory] (executor-thread-1) AppConfig = <null>
2020-05-14 15:54:41,126 INFO  [org.acm.res.RequestFilter] (executor-thread-1) SomeBean = <org.acme.restclient.SomeBean@75863ffd>
2020-05-14 15:54:41,127 INFO  [org.acm.res.RequestFilter] (executor-thread-1) AppConfig = <org.acme.restclient.AppConfig@7d71fc68>
```

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: Linux garkenbout-laptop-quintor 5.0.0-38-generic #41-Ubuntu SMP Tue Dec 3 00:27:35 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
 - Output of `java -version`: openjdk version "11.0.7" 2020-04-14
OpenJDK Runtime Environment 18.9 (build 11.0.7+10)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.7+10, mixed mode)
 - GraalVM version (if different from Java): didn't check
 - Quarkus version or git rev: 1.4.2.Final
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): Apache Maven 3.6.2 (40f52333136460af0dc0d7232c0dc0bcf0d9e117; 2019-08-27T17:06:16+02:00)
Maven home: /home/garkenbout/.m2/wrapper/dists/apache-maven-3.6.2-bin/795eh28tki48bv3l67maojf0ra/apache-maven-3.6.2
Java version: 11.0.7, vendor: Oracle Corporation, runtime: /home/garkenbout/.sdkman/candidates/java/11.0.7-open
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.0.0-38-generic", arch: "amd64", family: "unix"

### PR DESCRIPTION
**Pull request by mkouba**
- resolves #9301

CC @famod

### CODE REVIEW

**Review by phillip-kruger — APPROVED** [review]
LGTM

**Review by geoand — APPROVED** [review]

File: extensions/rest-client/deployment/src/main/java/io/quarkus/restclient/deployment/RestClientProcessor.java [hunk 1]
```diff
@@ -421,6 +423,15 @@ void registerProviders(BuildProducer<ReflectiveClassBuildItem> reflectiveClass,
                     .produce(new ReflectiveClassBuildItem(false, false, annotationInstance.value().asClass().toString()));
         }
 
+        // Register @RegisterClientHeaders for reflection
+        for (AnnotationInstance annotationInstance : index.getAnnotations(REGISTER_CLIENT_HEADERS)) {
+            AnnotationValue value = annotationInstance.value();
+            if (value != null) {
+                reflectiveClass
+                        .produce(new ReflectiveClassBuildItem(false, false, annotationInstance.value().asClass().toString()));
```
Comment (gsmet) [inline extensions/rest-client/deployment/src/main/java/io/quarkus/restclient/deployment/RestClientProcessor.java:431]: I usually recommend to use `.name().toString()`. The main reason is that for class it's ok but for types the toString() can lead to a non valid class name. Thus I think it's better to get used to always use `name().toString()`.

Comment (mkouba) [reply extensions/rest-client/deployment/src/main/java/io/quarkus/restclient/deployment/RestClientProcessor.java:431]: Hm, that's a good point. I just copied the code and did not realize there is `ClassInfo.toString()`. However, for `ClassInfo` it should be safe, in contrast with `org.jboss.jandex.ClassType`...

Comment (geoand) [reply extensions/rest-client/deployment/src/main/java/io/quarkus/restclient/deployment/RestClientProcessor.java:431]: `ClassInfo.toString` is safe, although yeah, calling `.name()` first does remove any possible reason to doubt the code :)

**Comment by famod** [comment]
@gsmet & @mkouba Can this be added to 1.8.1?

**Pull request merged**